test/distributed/test_c10d_common.py

```diff
@@ -1466,6 +1466,10 @@ def _call_collective_with_varying_tensors(self, backend, collective, *args):
         # multi tensor collectives
         if collective == dist.all_gather:
             collective([tensor], tensor, *args)
+        elif collective == dist.reduce_scatter:
+            if backend != "gloo":
```

Comment:
with the dispatchable functionality, what would happen if user call reduce_scatter on CPU?

Reply:
Sorry, I missed this question. It will call the `reduce_scatter` based on the configured backend for `CPU` (we will set a backend for CPU by default, but users can configure this). This is yet to be implemented in this PR